src/redux/books/book.js

```diff
@@ -1,21 +1,90 @@
 // Actions
 const ADDBOOK = 'bookstore/bookReducer/ADD_BOOK';
 const REMOVEBOOK = 'bookstore/bookReducer/REMOVE_BOOK';
+const APIBOOK = 'bookstore/bookReducer/API_BOOK';
+
+let initialState = [];
 
 // Actions Creators
-export const addBook = (newBook) => ({ type: ADDBOOK, payload: newBook });
+export const postBook = (newBook) => async () => {
+  await fetch('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/opIvy7T4p9EOFrXXjvZD/books', {
+    method: 'POST',
+    headers: {
+      accept: 'application/json',
+      'content-type': 'application/json',
+    },
+    body: JSON.stringify({
+      item_id: newBook.id,
+      title: newBook.title,
+      author: newBook.author,
+      category: 'my-category',
+    }),
+  });
+};
+
+export const removeBook = (id) => ({ type: REMOVEBOOK, payload: id });
+
+export const apiBook = (data) => ({
+  type: APIBOOK,
+  payload: data,
+});
 
-export const removeBook = (myBook) => ({ type: REMOVEBOOK, payload: myBook });
+export const getBooks = () => async (dispatch) => {
+  const itemUrl = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/opIvy7T4p9EOFrXXjvZD/books';
+  const response = await fetch(itemUrl);
+  const data = await response.json();
+  const data1 = [];
+  // for (const [key, value] of Object.entries(data))
+  Object.entries(data).forEach(([key, value]) => {
+    const newValue = value[0];
+    const newdata = {
+      ...newValue,
+      id: key,
+    };
+    data1.push(newdata);
+  });
+
+  // for (const element in data) {
+  //   const newdata = data[element][0];
+  //   const newdata2 = {
+  //     ...newdata,
+  //     id: element,
+  //   };
+  //   data1.push(newdata2);
+  // }
+  console.log(data);
+  console.log(data1);
```

Comment:
Please ensure that all commented codes and `console.log` are removed from your codebase. Commented code makes your code dirty.